yt_idefix/_io/commons.py

```diff
@@ -47,6 +47,54 @@ def padded(self) -> Coordinates:
         return Coordinates(arrs[0], arrs[1], arrs[2], self.array_shape)
 
 
+def mapFromCart(
+    xcart: np.ndarray,
+    ycart: np.ndarray,
+    zcart: np.ndarray,
+    geometry: str,
+) -> list[np.ndarray]:
+    shape = Shape(*xcart.shape)
+
+    # Reconstruct the polar coordinate system
+    if geometry == "polar":
+        # on disk coordinates are cell face coordinates.
+        # We need to convert from cartesian to polar,
+        # but no interpolation is needed.
+        r = np.sqrt(xcart[:, 0, 0] ** 2 + ycart[:, 0, 0] ** 2)
+        theta = np.unwrap(np.arctan2(ycart[0, :, 0], xcart[0, :, 0]))
+        z = zcart[0, 0, :]
+
+        coords = [r, theta, z]
+    else:
+        assert geometry == "spherical"
```

Comment:
Let's be extra careful here since all geometries are not covered yet
```suggestion
    elif geometry == "spherical":
```
... and add the following after this block
```python
    else:
        raise NotImplementedError
```

Reply:
Implemented!